python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -451,3 +458,110 @@ def close(self) -> None:
         if self.is_local_node(self._reader_node_id):
             self.ensure_registered_as_reader()
         self._worker.core_worker.experimental_channel_set_error(self._reader_ref)
+
+
+@PublicAPI(stability="alpha")
+class MultiChannel(ChannelInterface):
+    """
+    Can be used to send data to different readers via different channels.
+    For example, if the reader is in the same worker process as the writer,
+    the data can be sent via IntraProcessChannel. If the reader is in a different
+    worker process, the data can be sent via shared memory channel.
+
+    Args:
+        writer: The actor that may write to the channel. None signifies the driver.
+        readers: The actors that may read from the channel. None signifies
+            the driver.
+    """
+
+    def __init__(
+        self,
+        writer: Optional[ray.actor.ActorHandle],
+        readers: List[Optional[ray.actor.ActorHandle]],
+        _channel_dict: Optional[Dict[ray.ActorID, ChannelInterface]] = None,
+        _channels: Optional[Set[ChannelInterface]] = None,
+    ):
+        self._writer = writer
+        self._readers = readers
+        self._writer_registered = False
+        self._reader_registered = False
+        # A dictionary that maps the actor ID to the channel object.
+        self._channel_dict = _channel_dict or {}
+        # A set of channel objects. The set is a deduplicated version of _channel_dict.
```

Comment:
Improve the doc: it is the set of _channel_dict values?

Reply:
Updated https://github.com/ray-project/ray/pull/45591/commits/f49593c0e3fe262464f8307e5d29cfe0d63b85b1